lib/widgets/message_list.dart

```diff
@@ -395,9 +396,14 @@ class MarkAsReadWidget extends StatelessWidget {
             padding: const EdgeInsets.all(10),
             child: FilledButton.icon(
               style: FilledButton.styleFrom(
-                padding: const EdgeInsets.all(10),
-                textStyle: const TextStyle(fontSize: 18, fontWeight: FontWeight.w200),
-                shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(5)),
+                backgroundColor: const HSLColor.fromAHSL(1,227,0.78,0.59).toColor(),
+                fixedSize: const Size.fromHeight(38),
```

Comment:
This seems to have no effect.

Looking with the inspector:
* The button's height as a tap target is 48px. That's because Flutter's Material implementation is providing that behavior for us automatically, which is handy.
* Its height as it participates in layout is also 48px. That's fine but probably means we want to reduce the further padding we give it above and below.
* The height of the visible button area is 40px. That reflects the default for `ButtonStyle.minimumSize` for `FilledButton`.

So we can override `minimumSize` here to get the visible height to 38px. The 48px for the touchable height is good and we can leave that as it is. But then the surrounding padding should be adjusted to reflect that the `FilledButton` already contains some padding for the touch target.

Reply:
From what I can tell setting `fixedSize` would have the same visual effect as setting `minimumSize`, the logic in [_ButtonStyleState.build](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/button_style_button.dart#L320) shows it as being the final decision on the constraints of the button size (as does poking around the inspector and tweaking the values there). On reflection, it would be safer to set a minimum instead in case the button contents expand beyond the button and it is unconstrained to grow vertically. In any case, switched to `mininumSize` and reduced the containing vertical padding by a total of 10px.

Reply:
Well, I started from just an empirical observation: this line with `fixedSize` had no effect. If I commented it out, nothing changed. Then the further empirical observation is that the inspector shows the height of the visible button area as 40px, which is more than 38px.

OTOH when I tried setting `minimumSize` here to a small value, that did have an effect — the button got shorter. So `minimumSize` does have an effect where `fixedSize` doesn't, even if the code that implements that is a bit tangly.

Looking at the linked code, we have:
```dart
    final Size? resolvedMinimumSize = resolve<Size?>((ButtonStyle? style) => style?.minimumSize);
    final Size? resolvedFixedSize = resolve<Size?>((ButtonStyle? style) => style?.fixedSize);
    // …
    BoxConstraints effectiveConstraints = resolvedVisualDensity.effectiveConstraints(
      BoxConstraints(
        minWidth: resolvedMinimumSize!.width,
        minHeight: resolvedMinimumSize.height,
        maxWidth: resolvedMaximumSize!.width,
        maxHeight: resolvedMaximumSize.height,
      ),
    );
    if (resolvedFixedSize != null) {
      final Size size = effectiveConstraints.constrain(resolvedFixedSize);
      // … width …
      if (size.height.isFinite) {
        effectiveConstraints = effectiveConstraints.copyWith(
          minHeight: size.height,
          maxHeight: size.height,
        );
      }
    }
```

So `minimumSize` — after resolving a value for it using the defaults, which here means 40px height — feeds into `effectiveConstraints`, and then that meets `fixedSize` in this line:
```dart
      final Size size = effectiveConstraints.constrain(resolvedFixedSize);
```

When they disagree, then, which one wins? The answer is up to that `BoxConstraints.constrain` method. Its [docs say](https://api.flutter.dev/flutter/rendering/BoxConstraints/constrain.html):
> Returns the size that both satisfies the constraints and is as close as possible to the given size.

So the resulting size *will* satisfy the constraints; it may or may not agree with the size, and will only be as close to the given size as the constraints allow. The constraints win any disagreement.

Here, that means `effectiveConstraints`, and through it `minimumSize`, prevail over `fixedSize`.

Reply:
I see now, and had a closer inspection of the code and `BoxContraints.constrain` and stand (massively) corrected. I must have only tested with sizes > 40 as well? I did see a change as I adjusted the value but perhaps I didn't test with smaller values.